templates/ecs.yaml

```diff
@@ -0,0 +1,317 @@
+    ConstraintDescription: Please choose a valid instance type.
+  DockerImageTag:
+    Type: String
+    Description: The Docker image tae to run
+Mappings:
+  AWSRegionToAMI:
+    us-east-1:
+      AMIID: ami-0669eafef622afea1
+    us-west-2:
+      AMIID: ami-05d33de432484ef34
+Resources:
+  # iAtlas API task execution IAM role
+  # This is necessary so it can read the Secrets Manager entry that contains the external
+  # registry credentials
+  IAtlasIAMExecutionRole:
+    Type: AWS::IAM::Role
+    Properties:
+      AssumeRolePolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service:
+                - ecs-tasks.amazonaws.com
+            Action:
+              - 'sts:AssumeRole'
+      Path: /
+      Policies:
+        - PolicyName: AccessRegCredSecret
+          PolicyDocument:
+            Version: 2012-10-17
+            Statement:
+              - Effect: Allow
+                Action:
+                  - ssm:GetParameters
+                  - secretsmanager:GetSecretValue
+                  - kms:Decrypt
+                Resource:
+                  - !Sub 'arn:aws:ssm:${AWS::Region}:${AWS::AccountId}:parameter/*'
+                  - !Sub 'arn:aws:secretsmanager:${AWS::Region}:${AWS::AccountId}:secret:*'
+                  - !Sub 'arn:aws:kms:${AWS::Region}:${AWS::AccountId}:key/*'
+              - Effect: Allow
+                Action:
+                  - logs:CreateLogStream
+                Resource: '*'
+  ECSCluster:
+    Type: AWS::ECS::Cluster
+
+  # Security group for ECS nodes
+  # Should allow ephemeral ports from the ALB SG
+  # Should allow SSH from the Bastion hosts security group
+  EcsSecurityGroup:
+    Type: AWS::EC2::SecurityGroup
+    Properties:
+      GroupDescription: ECS Node Security Group
+      VpcId: !Ref 'VpcId'
+  EcsSecurityGroupALBports:
+    Type: AWS::EC2::SecurityGroupIngress
+    Properties:
+      GroupId: !Ref 'EcsSecurityGroup'
+      IpProtocol: tcp
+      FromPort: 31000
+      ToPort: 61000
+      SourceSecurityGroupId: !Ref 'EcsAlbSecurityGroup'
+  EcsSecurityGroupSSHfromBastion:
+    Type: AWS::EC2::SecurityGroupIngress
+    Properties:
+      GroupId: !Ref 'EcsSecurityGroup'
+      IpProtocol: tcp
+      FromPort: 22
+      ToPort: 22
+      SourceSecurityGroupId: !Ref 'BastionHostSgId'
+
+  # Security group for the ALB that goes in front
+  # Should only allow port 80 from the world for now
+  EcsAlbSecurityGroup:
+    Type: AWS::EC2::SecurityGroup
+    Properties:
+      GroupDescription: Load Balancer for ECS security group
+      VpcId: !Ref 'VpcId'
+  EcsAlbSecurityGroupHTTPinbound:
+    Type: AWS::EC2::SecurityGroupIngress
+    Properties:
+      GroupId: !Ref 'EcsAlbSecurityGroup'
+      IpProtocol: tcp
+      FromPort: 80
+      ToPort: 80
+      CidrIp: 0.0.0.0/0
+
+  CloudwatchLogsGroup:
+    Type: AWS::Logs::LogGroup
```

Comment:
i don't think this it is necessary to create this resource.  if your instance writes to cloudwatch you just need to give your instance role a cloudwatch policy to write to cloudwatch and it will automatically create and name the cloudwatch log group. Here's an example cloudwatch policy for you:
```
        -  PolicyDocument:
            Version: 2012-10-17
            Statement:
              - Effect: Allow
                Action:
                  - 'logs:CreateLogGroup'
                  - 'logs:CreateLogStream'
                  - 'logs:PutLogEvents'
                Resource: 'arn:aws:logs:*:*:*'
```